### PR TITLE
Follow up on object store selection PR.

### PR DESCRIPTION
**Pull request by jmchilton**
There are a bunch of non-functional changes that would modernize the object store selection PR https://github.com/galaxyproject/galaxy/pull/14073 - it is a project I would love to delegate but I'm going to work through them until the PR is merged in case that doesn't materialize.

- [x] Pydantic for object store API.
- [x] FastAPI for new user APIs
- [x] Pydanic for new user APIs
- [ ] Redo awkward target DOM stuff on HistorySelect component without breaking popover
- [ ] Migrate Vue components to TypeScript and composition API.
   - [x] SelectObjectStore.vue
   - [ ] client/src/components/Dataset/DatasetStorage/ObjectStoreRestrictionSpan.vue
   - [ ] client/src/components/History/CurrentHistory/HistoryPreferredStore.vue
   - [ ] client/src/components/ObjectStore/ObjectStoreBadge.vue
   - [ ] client/src/components/ObjectStore/ObjectStoreBadges.vue
   - [ ] client/src/components/ObjectStore/ShowSelectedObjectStore.vue
   - [ ] client/src/components/Tool/ToolTargetPreferredObjectStorePopover.vue
   - [ ] client/src/components/User/DiskUsage/Quota/ProvidedQuotaSourceUsageBar.vue
   - [ ] client/src/components/User/UserPreferredObjectStore.vue
   - [ ] client/src/components/Workflow/Run/WorkflowSelectPreferredObjectStore.vue
   - [ ] client/src/components/Workflow/Run/WorkflowStorageConfiguration.vue
   - [ ] client/src/components/Workflow/Run/WorkflowTargetPreferredObjectStorePopover.vue
- Migrate new axios and providers to fetches and (probably) pinia.
   - [x] objectstore info
   - [ ] quota stuff
- [x] Bug fix - consistent use of fa-hdd (see commit about why - but the original PR had a mix of fa-hdd and fa-database for objectstore selection).
- [x] Bug fix - only show icon in ... if selectable object stores are available in config.
    - [x] histories (bug fix)
    - [x] tools / users / workflows (already worked properly in original PR)
- Fine tuning “preferred” vs ... “selection” nomenclature in the UI
   - Automated default based on object store and job configuration stuff.
    - Allow admin to configure this if dynamic job runners are used.

Bigger projects:

- Framework for ObjectStore config validation/linting… 
- Connect object store linting into configuration linting.
- Create a UI for the object store linting.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by jmchilton** [comment]
Not done refining this functionality - but I think these are all atomic changes and fixes some presentational issues to give this functionality a better first impression... and the PR is green 😆  so I'm marking this as ready for review and I'll collect more issues and fixes other places.

File: lib/galaxy/webapps/galaxy/api/users.py [hunk 1]
```diff
@@ -74,11 +75,18 @@
 
 UserIdPathParam: DecodedDatabaseIdField = Path(..., title="User ID", description="The ID of the user to get.")
 APIKeyPathParam: str = Path(..., title="API Key", description="The API key of the user.")
+FlexibleUserIdPathParam: str = Path(..., title="User ID", description="The ID of the user to get or __current__.")
```
Comment (mvdbeek) [inline lib/galaxy/webapps/galaxy/api/users.py:81]: I think the type should be a union of EncodedDatabaseId and the literal current here.

File: client/src/components/History/CurrentHistory/HistoryCounter.vue [hunk 2]
```diff
@@ -12,19 +12,25 @@
...
-                    class="rounded-0 text-decoration-none"
-                    @click="showPreferredObjectStoreModal = true">
-                    <icon icon="hdd" />
-                </b-button>
-                <PreferredStorePopover
-                    :history-id="history.id"
-                    :history-preferred-object-store-id="historyPreferredObjectStoreId"
-                    :user="user">
-                </PreferredStorePopover>
+                <ConfigProvider v-slot="{ config }">
```
Comment (mvdbeek) [inline client/src/components/History/CurrentHistory/HistoryCounter.vue:15]: Can you use the `useConfig(true)` composable here please ? The ConfigProvider triggers a new fetch every time.

**Review by mvdbeek — APPROVED** [review]

**Comment by jmchilton** [comment]
More follow up issues are being tracked in https://github.com/galaxyproject/galaxy/issues/15692.

**Pull request merged**